gnovm/pkg/gnolang/values.go

```diff
@@ -1454,6 +1454,131 @@ func (tv *TypedValue) GetBigDec() *apd.Decimal {
 	return tv.V.(BigdecValue).V
 }
 
+// Sign returns -1, 0, or 1 depending on whether x < 0, x == 0, or x > 0;
+func (tv *TypedValue) Sign() int {
+	if tv.T == nil {
+		panic("type should not be nil")
+	}
+	switch tv.T.Kind() {
+	case IntKind:
+		v := tv.GetInt()
+		if v < 0 {
+			return -1
+		} else if v == 0 {
+			return 0
+		} else {
+			return 1
+		}
+	case Int8Kind:
+		v := tv.GetInt8()
+		if v < 0 {
+			return -1
+		} else if v == 0 {
+			return 0
+		} else {
+			return 1
+		}
+	case Int16Kind:
+		v := tv.GetInt16()
+		if v < 0 {
+			return -1
+		} else if v == 0 {
+			return 0
+		} else {
+			return 1
+		}
+	case Int32Kind:
+		v := tv.GetInt32()
+		if v < 0 {
+			return -1
+		} else if v == 0 {
+			return 0
+		} else {
+			return 1
+		}
+	case Int64Kind:
+		v := tv.GetInt64()
+		if v < 0 {
+			return -1
+		} else if v == 0 {
+			return 0
+		} else {
+			return 1
+		}
+	case UintKind:
+		v := tv.GetUint()
+		if v < 0 {
```

Comment:
I think the less than zero checks can be removed for unsigned types

Reply:
fixed.